haystack/utils/openai_utils.py

```diff
@@ -147,6 +148,26 @@ def openai_request(
         return response
 
 
+def check_openai_policy_violation(input: Union[List[str], str], headers: Dict) -> bool:
+    """
+    Calls the moderation endpoint to check if the text(s) violate the policy.
+    See [OpenAI Moderation API](https://platform.openai.com/docs/guides/moderation) for more details.
+    Returns true if any of the input is flagged as any of ['sexual', 'hate', 'violence', 'self-harm', 'sexual/minors', 'hate/threatening', 'violence/graphic'].
+    """
+    response = openai_request(url=OPENAI_MODERATION_URL, headers=headers, payload={"input": input})
+    results = response["results"]
+    flagged = any(res["flagged"] for res in results)
+    if flagged:
+        for result in results:
+            if result["flagged"]:
+                logger.debug(
+                    "OpenAI Moderation API flagged the text '%s' as a potential policy violation of the following categories: %s",
+                    input,
+                    result["categories"],
+                )
+    return flagged
```

Comment:
I was thinking about whether it makes sense to support more use cases by returning the result["categories"] here but users who need that info can also call `openai_request(url=OPENAI_MODERATION_URL, headers=headers, payload={"input": input})` themselves.

Reply:
I will add a flag to this function which will make it return a dict with the info when set. Although this would not be settable on the user level, it might be better to have it in the utilities just in case. What do you think?

Reply:
You can also leave it as is to keep this PR as simple as possible. As an alternative, I was thinking to always return a dict here. (no additional flag in the function). And that dict is either empty or not. If in doubt, I'd say let's keep it simple (as is). We can add more functionality later if needed.

Reply:
Alright, I changed it back to only return a simple bool 👍🏼